### PR TITLE
Circle.yml: JVM memory<=3.5GB, Gradle parallel+daemon builds

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -3,22 +3,15 @@ machine:
...
   environment:
     TERM: dumb
-    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
-
-general:
-  artifacts:
-    - error-handling/build/libs
-    - http-clients/build/libs
-    - http-servers/build/libs
-    - ssl-config/build/libs
```
Comment (markelliot) [inline circle.yml:13]: No desire to keep the artifacts?

File: circle.yml [hunk 2]
```diff
@@ -3,22 +3,15 @@ machine:
...
+    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx3400m -XX:+HeapDumpOnOutOfMemoryError"'
 
 dependencies:
   override:
     - ./gradlew resolveConfigurations
 
 test:
   override:
-    - ./gradlew build testBouncyCastle testSun
+    - ./gradlew --daemon --parallel build testBouncyCastle testSun
```
Comment (markelliot) [inline circle.yml:14]: Shouldn't we daemon everywhere if we daemon at all?

File: circle.yml (hunk 1 above)
Comment (uschi2000) [inline circle.yml:13]: As it stands, the artifact list was incomplete. Tests results and failures are collected automatically as far as I understand. Did you have another use for these?

File: circle.yml (hunk 2 above)
Comment (uschi2000) [inline circle.yml:14]: Yep, can do.

**Pull request merged**